src/rpc.rs

```diff
@@ -351,6 +409,9 @@ mod tests {
         let cluster_info = Arc::new(RwLock::new(
```

Comment:
This test is getting massive (about 3 times the size from before this PR). If you added a helper function that returned an initialized `cluster_info`, `io`, and `meta`, could you be able to break it up such that there's one test per request type with minimal code duplication?